Components/Helper/Currency.php

```diff
@@ -0,0 +1,262 @@
+<?php
+/**
+ * Copyright (c) 2018, Nosto Solutions Ltd
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ * this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its contributors
+ * may be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * @author Nosto Solutions Ltd <shopware@nosto.com>
+ * @copyright Copyright (c) 2018 Nosto Solutions Ltd (http://www.nosto.com)
+ * @license http://opensource.org/licenses/BSD-3-Clause BSD 3-Clause
+ */
+
+use Shopware_Plugins_Frontend_NostoTagging_Components_Account as NostoComponentAccount;
+use Shopware_Plugins_Frontend_NostoTagging_Bootstrap as Bootstrap;
+use Nosto\Operation\UpdateSettings as NostoUpdateSettings;
+use Nosto\Object\Settings as NostoSettings;
+use Nosto\Object\ExchangeRateCollection;
+use Nosto\Object\Signup\Account;
+use Nosto\Object\ExchangeRate;
+use Shopware\Models\Shop\Shop;
+use Shopware\Models\Shop\Currency;
+use Nosto\Object\Format;
+
+/**
+ * Helper class for Currency
+ *
+ * @package Shopware
+ * @subpackage Plugins_Frontend
+ */
+class Shopware_Plugins_Frontend_NostoTagging_Components_Helper_Currency
+{
+    const CURRENCY_SYMBOL_LEFT = 32;
+    const CURRENCY_SYMBOL_RIGHT = 16;
+    const CURRENCY_SYMBOL_DEFAULT = 0;
+    const CURRENCY_DECIMAL_CHAR = ',';
+    const CURRENCY_GROUPING_CHAR = '.';
+    const CURRENCY_DECIMAL_PRECISION = 2;
+
+    /**
+     * Update exchange rates for the given shop.
+     *
+     * @param Account $account
+     * @param Shop $shop
+     * @return bool
+     */
+    public static function updateCurrencyExchangeRates(Account $account, Shop $shop)
```

Comment:
This should not be in a helper. How about introducing a settings operation and put all settings & exchange related stuff there?

Reply:
Fixed, Thanks